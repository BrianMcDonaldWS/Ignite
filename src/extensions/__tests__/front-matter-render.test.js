import render from '../front-matter-render';

test('front-matter-render - simple author', () => {
  expect(
    render([
      {
        markup: `---\ntitle: First Post\nauthor: Andrew Lisowski\n---`
      }
    ])
  ).toMatchSnapshot();
});

test('front-matter-render - complex author', () => {
  expect(
    render([
      {
        markup: `---\ntitle: First Post\nauthor:\n  name: Andrew Lisowski\n  url: https://github.intuit.com/alisowski\n  email: lisowski54@gmail.com\n---`
      }
    ])
  ).toMatchSnapshot();
});

test('front-matter-render - banner image', () => {
  expect(
    render([
      {
        markup: `---\ntitle: First Post\nimage: https://pbs.twimg.com/profile_banners/7547562/1516584955/1500x500\n---`
      }
    ])
  ).toMatchSnapshot();
});

test('front-matter-render - error', () => {
  expect(() =>
    render([
      {
        markup: `---\ntitle= First Post\nauthor: Andrew Lisowski\n---`
      }
    ])
  ).toThrow();
});
