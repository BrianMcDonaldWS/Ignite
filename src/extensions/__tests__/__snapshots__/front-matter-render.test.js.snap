// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`front-matter-render - banner image 1`] = `
"
    <div class=\\"media blogHeader\\">
      <p id=\\"background-image\\">
        https://pbs.twimg.com/profile_banners/7547562/1516584955/1500x500
      </p>
      <div class=\\"media-content has-text-centered\\">
        

        <p class=\\"title blogTitle\\">
          First Post
        </p>
        <p class=\\"subtitle is-6 blogSubtitle\\">
          undefined
        </p>
      </div>
    </div>
  "
`;

exports[`front-matter-render - complex author 1`] = `
"
    <div class=\\"media blogHeader\\">
      <p id=\\"background-image\\">
        undefined
      </p>
      <div class=\\"media-content has-text-centered\\">
        <img src=\\"//www.gravatar.com/avatar/ff725a631f869cdb78beeb003825bb40\\" class=\\"authorImage\\" />

        <p class=\\"title blogTitle\\">
          First Post
        </p>
        <p class=\\"subtitle is-6 blogSubtitle\\">
          <a target=\\"_blank\\" href=\\"https://github.intuit.com/alisowski\\">Andrew Lisowski</a>
        </p>
      </div>
    </div>
  "
`;

exports[`front-matter-render - simple author 1`] = `
"
    <div class=\\"media blogHeader\\">
      <p id=\\"background-image\\">
        undefined
      </p>
      <div class=\\"media-content has-text-centered\\">
        

        <p class=\\"title blogTitle\\">
          First Post
        </p>
        <p class=\\"subtitle is-6 blogSubtitle\\">
          Andrew Lisowski
        </p>
      </div>
    </div>
  "
`;
